recipes/icontract/meta.yaml

```diff
@@ -0,0 +1,45 @@
+{% set name = "icontract" %}
+{% set version = "2.6.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/icontract-{{ version }}.tar.gz
+  sha256: 1ef9332ab4ca10f4da0370cec3743b88e10f8aac3cfcb9bc528008e1b2f09fae
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - asttokens >=2,<3
+    - python >=3.6
+    - typing_extensions
+
+test:
+  imports:
+    - icontract
+    - tests_3_6
+    - tests_3_7
+    - tests_3_8
```

Comment:
If this is passing, that means it's installing the test as a test package/module in the global package namespace, which is not good and can become a problem since they will pollute the end-user environment. This is not a problem that you can necessarily solve yourself but can you ask upstream to fix this, please? Test folders should never be installed as a separate package.

Reply:
Great catch!
I raised an issue [0] upstream. How would you suggest to proceed?

[0] https://github.com/Parquery/icontract/issues/239

Reply:
The best way would be to add a PR to upstream fixing this and then including the PR as a [patch](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#patches) here.

Alternatively, you can also add some sort of script to the build section that removes it but that's really a dirty/last resort solution.

Reply:
The upstream problem is fixed and a patched version already released on pypi. I updated the recipe to this new version on pypi: [85e458c](https://github.com/conda-forge/staged-recipes/pull/18031/commits/85e458cb0ac1ad2dad06fc4696adb7b106c0ce88)

Do I understand correctly that we no longer need aforementioned approach?

Reply:
Yes, it's not required anymore. From my side this can be merged.